### ISSUE TITLE
Crash when loading a vocabulary

### ISSUE THREAD

**Issue by jneubert** [jrvosse/amalgame#11]
In a newly set up instance of amalgame (5fc88e3), the instance crashes when I try to load a small vocabulary from a local file. Last log entry is:

/*Fri Apr 10 13:00:59 2015*/ request(54, 1428663659.506, [session('adff-d548-4689-2b7c.s279606780'),peer(ip(127,0,0,1)),method(post),request_uri('/amalgame-sdmx/servlets/uploadData'),path('/amalgame-sdmx/servlets/uploadData'),http_version(1-1),host('127.0.0.1'),port(30202),user_agent('Mozilla/5.0 (Windows NT 6.1; WOW64; rv:36.0) Gecko/20100101 Firefox/36.0'),dnt('1'),referer('http://111.111.111.111/amalgame-sdmx/user/loadFile'),cookie([swipl_session='adff-d548-4689-2b7c.s279606780']),content_type('multipart/form-data; boundary=---------------------------562544013346'),x_forwarded_for('134.245.95.1'),x_forwarded_host('111.111.111.111'),x_forwarded_server('example.com'),connection('Keep-Alive'),content_length(26264)]).

(Addresses obfuscated).

Maybe this is somehow related to #10

**Comment by jneubert** [comment]
I tried this again with a newly installed swi-pl (7.1.36) and a newly installed Cliopatria (via git clone). When I run Cliopatria on the console, I get one and only one message on the loading via http attempt., before the program terminates:
```
?- debug(ag_expand).
true.

?- /usr/local/bin/swipl: symbol lookup error: /usr/local/lib/swipl-7.1.36/lib/x86_64-linux/archive4pl.so: undefined symbol: archive_read_set_callback_data
```
This correspondes with a message from the swi-pl build:
```
% Loading library(archive) ..................... ok
/usr/local/bin/swipl: symbol lookup error: /usr/local/lib/swipl-7.1.36/lib/x86_64-linux/archive4pl.so: undefined symbol: archive_read_set_callback_data
make[1]: *** [check-installation] Error 127
make[1]: Leaving directory `/usr/local/src/pl-7.1.36/src'
make: *** [check-installation] Error 2

QA Notice: Package triggers severe warnings which indicate that it
           may exhibit random runtime failures.
archive4pl.c:724: warning: implicit declaration of function ‘archive_read_set_callback_data’
archive4pl.c:725: warning: implicit declaration of function ‘archive_read_set_open_callback’
archive4pl.c:726: warning: implicit declaration of function ‘archive_read_set_read_callback’
archive4pl.c:727: warning: implicit declaration of function ‘archive_read_set_skip_callback’
archive4pl.c:728: warning: implicit declaration of function ‘archive_read_set_seek_callback’
archive4pl.c:729: warning: implicit declaration of function ‘archive_read_set_close_callback’
archive4pl.c:731: warning: implicit declaration of function ‘archive_read_open1’

No errors during package build
% Loading library(archive) ..................... ok
/usr/local/bin/swipl: symbol lookup error: /usr/local/lib/swipl-7.1.36/lib/x86_64-linux/archive4pl.so: undefined symbol: archive_read_set_callback_data
make[1]: *** [check-installation] Error 127
make[1]: Leaving directory `/usr/local/src/pl-7.1.36/src'
make: *** [check-installation] Error 2

QA Notice: Package triggers severe warnings which indicate that it
           may exhibit random runtime failures.
archive4pl.c:724: warning: implicit declaration of function ‘archive_read_set_callback_data’
archive4pl.c:725: warning: implicit declaration of function ‘archive_read_set_open_callback’
archive4pl.c:726: warning: implicit declaration of function ‘archive_read_set_read_callback’
archive4pl.c:727: warning: implicit declaration of function ‘archive_read_set_skip_callback’
archive4pl.c:728: warning: implicit declaration of function ‘archive_read_set_seek_callback’
archive4pl.c:729: warning: implicit declaration of function ‘archive_read_set_close_callback’
archive4pl.c:731: warning: implicit declaration of function ‘archive_read_open1’

No errors during package build
```

**Comment by jneubert** [comment]
Caused by mismatch: Centos 6.6 offers libarchive 2.8.3, while Cliopatria requires 3.1.2.

With the help of Jan, the issue could be solved by just deleting /usr/local/lib/swipl-7.1.36/library/archive.pl, which causes Cliopatria to fall back to a prior mechanism.
